tasks/yesno.py

```diff
@@ -0,0 +1,301 @@
+from keras.regularizers import l2
+from keras.layers.embeddings import Embedding
+from keras.models import Graph
+from pysts.clasrel_layers import Reshape_, WeightedMean
+import pysts.nlp as nlp
+
+from . import AbstractTask
+
+
+class Container:
+    """Container for merging questions together."""
+    def __init__(self, q_text, s0, s1, si0, si1, f0, f1, y):
+        self.q_text = q_text  # str of question
+        self.s0 = s0
+        self.s1 = s1
+        self.si0 = si0
+        self.si1 = si1
+        self.f0 = f0
+        self.f1 = f1
+        self.y = y
+
+
+
+class YesNoTask(AbstractTask):
+    def __init__(self):
+        self.name = 'yesno'
+        self.emb = None
+        self.vocab = None
+
+    def config(self, c):
+        c['task>model'] = True
+        c['loss'] = 'binary_crossentropy'
+        c['max_sentences'] = 50
+        c['spad'] = 60
+        c['embdim'] = 50
+        c['nb_epoch'] = 30
+
+        c['opt'] = 'adam'
+        c['inp_e_dropout'] = 0.
+        c['dropout'] = 0.
+        c['l2reg'] = 0.01
+        c['e_add_flags'] = True
+        c['ptscorer'] = mlp_ptscorer
+        c['mlpsum'] = 'sum'
+        c['Ddim'] = .1
+        c['loss'] = 'binary_crossentropy'
+        c['nb_epoch'] = 100
+        c['batch_size'] = 10
+        c['class_mode'] = 'binary'
+
+        # old rnn
+        c['pdim'] = 2.5
+        c['pact'] = 'tanh'
+
+    def load_set(self, fname, cache_dir=None):
+        # TODO: Make the cache-handling generic,
+        # and offer a way to actually pass cache_dir
+        save_cache = False
+        if cache_dir:
+            import os.path
+            fname_abs = os.path.abspath(fname)
+            from hashlib import md5
+            cache_filename = "%s/%s.p" % (cache_dir, md5(fname_abs.encode("utf-8")).hexdigest())
+            try:
+                with open(cache_filename, "rb") as f:
+                    return pickle.load(f)
+            except (IOError, TypeError, KeyError):
+                save_cache = True
+
+        s0, s1, y, kw, akw, t = loader.load_anssel(fname, skip_oneclass=False)
+
+        if self.vocab is None:
+            vocab = Vocabulary(s0 + s1)  # FIXME: lower?
+        else:
+            vocab = self.vocab
+
+        si0 = vocab.vectorize(s0, spad=self.c['spad'])
+        si1 = vocab.vectorize(s1, spad=self.c['spad'])
+        f0, f1 = nlp.sentence_flags(s0, s1, self.c['spad'], self.c['spad'])
+        gr = graph_input_anssel(si0, si1, y, f0, f1, s0, s1, kw=kw, akw=akw)
+        gr, y = self.merge_questions(gr)
+        if save_cache:
+            with open(cache_filename, "wb") as f:
+                pickle.dump((s0, s1, y, vocab, gr), f)
+                print("save")
+
+        return (gr, y, vocab)
+
+    def build_model(self, module_prep_model, do_compile=True):
+
+        model = build_model(self.emb, self.vocab, module_prep_model, self.c)
+
+        # for lname in self.c['fix_layers']:  # whats this??
+        #     model.nodes[lname].trainable = False
```

Comment:
What's the reason to comment this out and not to support fixing some layers in place?